machine_learning/similarity_search.py

```diff
@@ -0,0 +1,109 @@
+"""
+Simularity search is a search algorithm for finding the nearest vector from
+vectors, used in natural language processing.
+In this algorithm, it calculates distance with euclidean distance and
+returns a list containing two data for each vector:
+    1. the nearest vector
+    2. distance between the vector and the nearest vector
+"""
+import numpy as np
+import math
+
+
+def euclidean(input_a, input_b):
+    """
+    Calculates euclidean distance between two data. The result should be float.
+    >>> euclidean(0, 1)
+    1.0
+    >>> euclidean(np.array([0, 1]),np.array([1, 1]))
+    1.0
+    >>> euclidean(np.array([0, 0, 0]), np.array([0, 0, 1]))
+    1.0
+    """
+    dist = 0
+
+    if type(input_a) == type(input_b):
+        if type(input_a) != np.ndarray:
+            dist = pow(input_a - input_b, 2)
+        else:
+            for index in range(len(input_a)):
+                dist += pow(input_a[index] - input_b[index], 2)
+        return math.sqrt(dist)
+    return None
+
+
+def similarity_search(dataset: np, value: np) -> list:
```

Comment:
These type hints seem off. Do the arguments `dataset` and `value` require the numpy module type? This should probably changed to:
```python
def similarity_search(dataset: np.ndarray, value: np.ndarray) -> list:
    ...
```